app.js

```diff
@@ -1,63 +1,225 @@
-      var value = req.query.ports;
-      if (!Array.isArray(value)) {
-        value = [value];
-      }
-      options.ports = value;
-    }
-    options.headerText = options.status;
-    if (options.status) {
-      if (options.status === 'buildFailed') {
-        options.headerText = 'build failed.';
-        options.status = 'failed to build';
-      } else {
-        options.status = 'is ' + options.status;
-      }
+  }
+
+  if (req.query.type === 'signin') {
+    log.trace('processNaviError type signin');
+    return res.render('pages/signin', options);
+  }
+  if (req.query.type === 'not_running') {
+    log.trace('processNaviError type not_running');
+
+    // container state error pages.
+    // - Not running (building, starting, crashed)
+    // - Running, but unresponsive
+    var status = req.instance.status();
+    log.trace({
+      status: status,
+      options: options
+    }, 'processNaviError instance status');
+
+    options.status = status;
+    switch(status) {
+      case 'stopped':
+      case 'crashed':
+      case 'stopping':
+        options.headerText = 'is ' + status;
+        res.render('pages/dead', options);
+        break;
+      case 'running':
+        // The instance could have started after Navi fetched it and proxied to detention.
+        // Might not be the best idea to trigger a refresh, could easily result in user-unfriendly
+        // infinite redirect loops. Better to display an error page prompting user to refresh?
+        options.headerText = 'is running';
+        res.render('pages/dead', options);
+        break;
+      case 'buildFailed':
+        options.headerText = 'build failed';
+        res.render('pages/dead', options);
+        break;
+      case 'building':
+        options.headerText = 'is building';
+        res.render('pages/dead', options);
+        break;
+      case 'neverStarted':
+      case 'starting':
+        options.headerText = 'is starting';
+        res.render('pages/dead', options);
+        break;
+      case 'unknown':
+        options.headerText = 'unknown';
+        res.render('pages/dead', options);
+        break;
     }
-    res.render('pages/' + page, options);
-  } else {
-    res.render('pages/invalid', options);
+    return;
   }
-});
+  if (req.query.type === 'ports'){
+    log.trace('processNaviError type ports');
+    /*
+     * Currently not implemented, might be bundled into 'unresponsive'
+     *
+     * Userland hipache will only route to navi if a request is made to an elastic url on a port
+     * that's explicitly set on the instance (we set hipache redis entries when ports are exposed)
+     * otherwise userland-hipache will return an error page due to a lack of a redis entry.
+     *
+     * Probably could fix by patching Hipache or perhaps reading the manual to see if there's a
+     * some kind of forward-for-all-ports functionality
+     *
+     * Anand if you read this Monday morning lets chat about it at 3pm
+     */
```

Comment:
The unresponsive page will show you a list of ports. That's probably what you've been seeing. We don't have anything yet that can display a message if you try to access a port that isn't set as exposed on the instance document